pkg/crdutil/crdutil.go

```diff
@@ -14,6 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+//nolint:depguard,lll
```

Comment:
```
pkg/crdutil/crdutil.go:62: the line is 148 characters long, which exceeds the maximum of 120 characters. (lll)
		"Process the directory used in -f, --filename recursively. Useful when you want to manage related manifests organized within the same directory.")
pkg/crdutil/crdutil.go:29:2: import 'github.com/spf13/pflag' is not allowed from list 'main' (depguard)
	"github.com/spf13/pflag"
	^
make: *** [lint] Error 1
```

IMO both is fine to ignore

Reply:
But if we say that we don't want to have the direct dependency to pflag I can remove it and use stdlib flag

Reply:
`lll` can be removed by splitting the string in multiple lines. I think that's easy to do.

For the `depguard`, maybe worth adding it here instead https://github.com/NVIDIA/k8s-operator-libs/blob/main/.golangci.yaml#L104.

In any case, if we can go with flag (like we used to), would be better to avoid yet another dependency.

Reply:
> lll can be removed by splitting the string in multiple lines. I think that's easy to do.

Maybe I can also remove the second sentence. It's just a copy from `kubectl`, but we can adjust it IMO.

> For the depguard, maybe worth adding it here instead https://github.com/NVIDIA/k8s-operator-libs/blob/main/.golangci.yaml#L104.

Yes, also thought about this, but we should rather discuss why we need this anyway?
All dependencies, if they are indirect or direct will be indirect dependencies in the consumers go.mod.
IMO we don't need that `depguard` check.

Repro with the following:

```
❯ ll
total 28K
-rw-r--r-- 1 tgiese wheel 2.8K Nov 27 14:19 go.mod
-rw-r--r-- 1 tgiese wheel  18K Nov 27 14:19 go.sum
-rw-r--r-- 1 tgiese wheel  119 Nov 27 14:16 main.go
❯ cat main.go
package main

import (
	"github.com/NVIDIA/k8s-operator-libs/pkg/crdutil"
)

func main() {
	crdutil.EnsureCRDsCmd()
}
```

https://github.com/NVIDIA/k8s-operator-libs/blob/92553f3478d75f72c6ad58a8b8b329866bed5275/go.mod#L9
require github.com/spf13/pflag v1.0.5

```
❯ go mod tidy; grep -e replace -e pflag go.mod
	github.com/spf13/pflag v1.0.5 // indirect
replace github.com/NVIDIA/k8s-operator-libs => github.com/tobiasgiese/k8s-operator-libs v0.0.0-20241127003902-92553f3478d7
```

and without the replace (i.e., with pflag indirect dep in the lib)
```
❯ go mod tidy; grep -e replace -e pflag go.mod
	github.com/spf13/pflag v1.0.5 // indirect
❯
```

Reply:
> All dependencies, if they are indirect or direct will be indirect dependencies in the consumers go.mod.
IMO we don't need that depguard check.

In general, it's good to avoid importing deps on packages that are imported by other consumers to avoid diamond dependencies. In that particular case, you proved that the `github.com/NVIDIA/k8s-operator-libs/pkg/crdutil` is importing some package that already imports pflag, so in that case, it's not that bad to import pflag. But this is not the case for any package I believe.

Reply:
Added `github.com/spf13/pflag` to the depguard allow list and cut the flag description